mdio/utils.h

```diff
@@ -131,18 +131,43 @@ Result<void> DeleteDataset(std::string dataset_path) {
     return dsRes.status();
   }
 
+  // As we cannot get to this point without a valid dataset, the input should be
+  // able to be considered safe.
+  // Failed sanitization is assumed to be caught by lower levels of Tensorstore.
+
+  // TODO(BrianMichell): This still isn't the safest assumption.
+
   // Delete the dataset
   try {
-    // TODO(BrianMichell): This is all probably a security risk...
     if (dataset_path.rfind("gs://", 0) == 0) {
       // Google Cloud Storage
-      std::system(("gsutil rm -r " + dataset_path).c_str());
+      int result = std::system(("gsutil rm -r " + dataset_path).c_str());
+      if (result != 0) {
+        return absl::InternalError(
+            "Failed to delete dataset on GCS with "
+            "exit code: " +
+            std::to_string(result));
+      }
     } else if (dataset_path.rfind("s3://", 0) == 0) {
       // Amazon S3
-      std::system(("aws s3 rm --recursive " + dataset_path).c_str());
+      int result =
+          std::system(("aws s3 rm --recursive " + dataset_path).c_str());
+      if (result != 0) {
+        return absl::InternalError(
+            "Failed to delete dataset on S3 with "
+            "exit code: " +
+            std::to_string(result));
+      }
     } else {
```

Comment:
We could add another check:


} else if (dataset_path.rfind("/", 0) == 0) {
   ...
} else {
return absl::InternalError(
            "Failed to delete.  Unknown dataset type for dataset:" +
            dataset_path);
      }

Reply:
I believe the local filesystem operations are safe. Also I'm against requiring absolute path and we need to support more than POSIX paths.